dj/api/helpers.py

```diff
@@ -31,12 +31,13 @@
 from dj.sql.parsing.backends.exceptions import DJParseException
 
 
-def get_node_by_name(
+def get_node_by_name(  # pylint: disable=too-many-arguments
     session: Session,
     name: str,
     node_type: Optional[NodeType] = None,
     with_current: bool = False,
     raise_if_not_exists: bool = True,
+    raise_if_exists: bool = False,
```

Comment:
nit and style opinion: I think we should avoid these raise_ args and let the calling logic catch the exception. Even when this is set to not raise we still have to check something like `if not foo:`. I realize it's painful to have to do try/excepts everywhere and feels duplicative since DJExceptions raise nice HTTP response, so maybe an alternative could be to just have separate functions for these different flows such as `raise_if_node_exists(...)`.

Reply:
I moved these to a separate `raise_if_node_exists` function. I can do a separate follow-up that changes this setup for all the `get_X_by_name` helpers.